ogr/ogrsf_frmts/geojson/ogrgeojsonutils.cpp

```diff
@@ -481,6 +484,60 @@ bool GeoJSONSeqIsObject(const char *pszText)
            IsLikelyNewlineSequenceGeoJSON(nullptr, nullptr, pszText);
 }
 
+/************************************************************************/
+/*                        JSONFGFileIsObject()                          */
+/************************************************************************/
+
+static bool JSONFGFileIsObject(GDALOpenInfo *poOpenInfo)
+{
+    // 6000 somewhat arbitrary. Based on other JSON-like drivers
+    if (poOpenInfo->fpL == nullptr || !poOpenInfo->TryToIngest(6000))
+    {
+        return false;
+    }
+
+    const char *pszText =
+        reinterpret_cast<const char *>(poOpenInfo->pabyHeader);
+    return JSONFGIsObject(pszText);
+}
+
+bool JSONFGIsObject(const char *pszText)
+{
+    if (!IsJSONObject(pszText))
+        return false;
+
+    const std::string osWithoutSpace = GetCompactJSon(pszText, strlen(pszText));
+
+    {
+        const auto nPos = osWithoutSpace.find("\"conformsTo\":[");
+        if (nPos != std::string::npos)
+        {
+            if (osWithoutSpace.find("\"[ogc-json-fg-1-0.1:core]\"", nPos) !=
+                    std::string::npos ||
+                osWithoutSpace.find(
+                    "\"http://www.opengis.net/spec/json-fg-1/0.1\"", nPos) !=
+                    std::string::npos)
+            {
+                return true;
+            }
+        }
+    }
+
+    if (osWithoutSpace.find("\"coordRefSys\":") != std::string::npos ||
```

Comment:
This applies to the "osWithoutSpace" variable which is JSON string with all those spaces/newlines removed.
Unfortunately there's no safe regexp library available in the standard C++ library (the existing one being considered for deprecation because of issues, and not implemented in all implementations)